design/secrets_provider_milestone1_design.md

```diff
@@ -42,54 +42,103 @@
+###### How this answers the requirements?
 
-**Decision:** Run as a Job
+For Milestone 1, running as a **Job** allows separation from apps deployment and serve multiple apps at once.
+When Secrets Provider finishes updating the K8s secrets, the **Job** completes successfully.
 
-At current, a "Job" is not an existing application identity granularity. In terms of granularities, a "Job" is at the same level as "Deployment". From our understanding customers do not regularly use "Deployment" as their chose granularity and tend to use "Service Account" / "Namespace". Because of this, support for a "Job" granularity might not provide value when customers can still use "Service Account" / "Namespace".
+For Milestone 2, running continuously as a **Deployment** or **Sidecar** allows listening on changes and handling them accordingly.
 
-We decided to run as a Job instead of Deployment because for Milestone 1 we are concerned with delivering Conjur values only once. At this time we are not supporting updating K8s Secrets with Conjur secret across multiple runs so deploying a Pod that will run continuously in the environment of a customer is a waste of their resources. Especially because the process of retrieving secrets will only happen once, at the inital spin up of the Secrets Provider.
+For Milestone 3, running by the app as **Init Container** or **Sidecar** allows writing and updating the secrets to files in a shared volume with the app.
 
-Additionally, when using Deployment to spin up a pod and the task finishes, the will container exit and be deleted. Now that the Secrets Provider needs to sit separately, when that container exists so too will the pod. Therefore, when using Deployment, Kuberentes will try to restart the Pod. After a few attempted restarts, the pod will error in a BackOff state. Because of this, we are unable to use Deployment to spin up a pod and must use a Job.
+###### What benefits does this solution provide?
+
+1. **Clarity**
+   Allowing customer to select the deployment to get the desired value
+2. **Maintainability**
+   Using one image for all variations simplifies code maintenance 
+3. **Reusability**
+   Using one image for all variations allows reuse of existing code across all variations
+4. **Extensibility**
+   Supporting various flows using the same solution opens the door for more secrets sources and targets
+
+###### What drawbacks does this solution have?
+
+**Job** is not an existing application identity granularity.
+As a result, one can define the host representing the Secrets Provider in Conjur policy using only **namepsace** and/or **service account** granularities.
```

Comment:
namepsace -> namespace

Reply:
isn't it true also for solutions? if not, how it'll change for other options?